src/main/kotlin/com/openlattice/organizations/ExternalDatabaseManagementService.kt

```diff
@@ -0,0 +1,369 @@
+        //remove all tables/columns within org
+        val belongsToDeletedDB = Predicates.equal("organizationId", orgId)
+        val tablesToDelete = organizationExternalDatabaseTables.values(belongsToDeletedDB)
+        tablesToDelete.forEach { table ->
+            organizationExternalDatabaseTables.remove(table.id)
+            aclKeyReservations.release(table.id)
+            val belongsToDeletedTable = Predicates.equal("tableId", table.id)
+            val columnsToDelete = organizationExternalDatabaseColumns.values(belongsToDeletedTable)
+            columnsToDelete.forEach { column ->
+                organizationExternalDatabaseColumns.remove(column.id)
+                aclKeyReservations.release(column.id)
+            }
+        }
+
+        //drop db from schema
+        val dbName = PostgresDatabases.buildOrganizationDatabaseName(orgId)
+        assemblerConnectionManager.connect(dbName).use {
+            it.connection.createStatement().use { stmt ->
+                stmt.execute("DROP DATABASE $dbName")
+            }
+        }
+    }
+
+    /*PERMISSIONS*/
+    fun addTrustedUser(orgId: UUID, userPrincipal: Principal, ipAdresses: Set<String>) {
+        val dbName = PostgresDatabases.buildOrganizationDatabaseName(orgId)
+        val userName = getDBUser(userPrincipal.id)
+        //edit the pg_hba file through some magic. must. become. magician.
+    }
+
+    /**
+     * Sets privileges for a user on an organization's table or column
+     */
+    fun executePrivilegesUpdate(action: Action, acls: List<Acl>) {
+        val aclsByOrg = acls.groupBy { it.aclKey[0] }
+        aclsByOrg.forEach { (orgId, acls) ->
+            val dbName = PostgresDatabases.buildOrganizationDatabaseName(orgId)
+            assemblerConnectionManager.connect(dbName).use { dataSource ->
+                dataSource.connection.autoCommit = false
+                dataSource.connection.createStatement().use { stmt ->
+                    acls.forEach {
+                        val tableAndColumnNames = getTableAndColumnNames(AclKey(it.aclKey))
+                        it.aces.forEach { ace ->
+                            val dbUser = getDBUser(ace.principal.id)
+                            val privileges = mutableListOf<String>()
+                            if (ace.permissions.contains(Permission.OWNER)) {
+                                privileges.add("ALL")
+                            } else {
+                                if (ace.permissions.contains(Permission.WRITE)) {
+                                    privileges.addAll(listOf("INSERT", "UPDATE"))
+                                }
+                                if (ace.permissions.contains(Permission.READ)) {
+                                    privileges.add("SELECT")
+                                }
+                            }
+                            val sql = createPrivilegesSql(action, privileges, tableAndColumnNames.first, tableAndColumnNames.second, dbUser)
```

Comment:
You should skip/throw error if `ace.permissions` do not contain any of the 3, because `privileges` will be empty. (or check if privileges are empty)

Reply:
gotcha ty!